skimage/filters/thresholding.py

```diff
@@ -766,27 +766,36 @@ def _mean_std(image, w):
            Retrieval XV, (San Jose, USA), Jan. 2008.
            DOI:10.1117/12.767755
     """
-    if w == 1 or w % 2 == 0:
+    if isinstance(w, int):
+        w1 = w
+        w2 = w
+        kern = np.zeros((w2 + 1,) * image.ndim)
+
+    if isinstance(w, tuple) and len(w) == image.ndim:
+        w1 = w[0] * w[1]
+        w2 = max(w)
+        kern = np.zeros(w)
+
+    if w1 == 1 or w1 % 2 == 0:
         raise ValueError(
             "Window size w = %s must be odd and greater than 1." % w)
 
-    left_pad = w // 2 + 1
-    right_pad = w // 2
+    left_pad = w2 // 2 + 1
+    right_pad = w2 // 2
     padded = np.pad(image.astype('float'), (left_pad, right_pad),
                     mode='reflect')
```

Comment:
Instead of padding by the biggest dimension, you need to pad each dimension differently. Check the documentation for `np.pad`.